### ISSUE TITLE
Host matching ACLs missing <host>:<port> form

### ISSUE THREAD

**Issue by bedis** [haproxytech/kubernetes-ingress#11]
Hi,

I think the ACLs matching Host headers for routing are inaccurate:
```use_backend service1 if { req.hdr(host) -i foo.bar.com } { path_beg /foo }```

Host header could have the following forms:
- foo.bar.com
- foo.bar.com:<port>

The acl above will not match the second form.
So it would be good to append to this acl, a new host header to which we append the port configured on the frontend

**Comment by ejez** [comment]
Could be solved by using hdr_reg instead of hdr:
```js
req.hdr_reg(host) -i ^foo\.bar\.com(:[0-9]+)?$
```
in file:  kubernetes-ingress/controller-backend-switching.go :
```go
import (
...
	"regexp"
...
)
...
				condTest = fmt.Sprintf("{ req.hdr_reg(host) -i %s } ", "^" + regexp.QuoteMeta(rule.Host) + "(:[0-9]+)?$")
...
```

However use of regex could slow down evaluating ACLs

**Comment by oktalz** [comment]
Hi, 
yes regex option is not the best option here.

i need to check it but maybe something like this would work
```
  acl acl_1 hdr_beg(host) -i foo.bar
  use_backend default-web-8181 if acl_1 { path_beg /web }
```

**Comment by bedis** [comment]
HAProxy proposes a few options:
- use regexes, but as you explained it's counter productive from a performance point of view and at scale
- use the hdr_dom comparison method which matches strings between '.' and ':'. But in such case, this would be equivalent to match *.foo.bar[:port]
- list all possible values in the current ACL: foo.bar foo.bar:80 or foo.bar:443 where it's relevant

The most accurate one is the latest one

**Comment by mikecurrancs** [comment]
Have ACLs been implemented in the build yet? I have an application to host, that appends :<port> to the host header, thereby not matching 

* use_backend service1 if { req.hdr(host) -i foo.bar.com } { path_beg /foo }

I do not see any documentation, but I do see code in controller-acls.go for ACL methods.

**Comment by Mo3m3n** [comment]
Hi
This was fixed from `v1.4.3` in the commit https://github.com/haproxytech/kubernetes-ingress/commit/aac95d5fef1377b26df9cde6d49efeb0b5a13001
Fell free to reopen this issue if this does not answer your requirement.
